src/praisonai-bot/praisonai_bot/scheduler/condition_gate.py

```
@@ -0,0 +1,143 @@
+            if proc is not None:
+                try:
+                    if _POSIX:
+                        os.killpg(os.getpgid(proc.pid), 9)
+                    else:  # pragma: no cover - non-POSIX
+                        proc.kill()
+                except (ProcessLookupError, PermissionError, OSError):  # pragma: no cover
+                    proc.kill()
+                try:
+                    proc.communicate(timeout=5)
+                except Exception:  # pragma: no cover - best-effort reap
+                    pass
```

Ruff (0.15.20)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)